modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/model/circuit.scala

```diff
@@ -50,20 +55,59 @@ object circuit {
   final case class StatusAndObserverFocus(isLogged: Boolean, name: Option[String], instrument: Instrument, id: Option[SequenceId], observer: Option[Observer], status: Option[SequenceState], targetName: Option[TargetName]) extends UseValueEq
   final case class StatusAndStepFocus(isLogged: Boolean, instrument: Instrument, id: Option[SequenceId], stepConfigDisplayed: Option[Int]) extends UseValueEq
   final case class StepsTableFocus(id: SequenceId, instrument: Instrument, state: SequenceState, steps: List[Step], stepConfigDisplayed: Option[Int], nextStepToRun: Option[Int]) extends UseValueEq
-  final case class ControlModel(id: SequenceId, isPartiallyExecuted: Boolean, nextStepToRun: Option[Int], status: SequenceState)
-  final case class SequenceControlFocus(isLogged: Boolean, isConnected: Boolean, control: Option[ControlModel])
+  object StepsTableFocus {
+    implicit val eq: Equal[StepsTableFocus] = Equal.equalA
+  }
+  final case class StepsTableAndStatusFocus(status: ClientStatus, stepsTable: Option[StepsTableFocus]) extends UseValueEq
+  final case class ControlModel(id: SequenceId, isPartiallyExecuted: Boolean, nextStepToRun: Option[Int], status: SequenceState) extends UseValueEq
+  final case class SequenceControlFocus(isLogged: Boolean, isConnected: Boolean, control: Option[ControlModel]) extends UseValueEq
+
+  /**
+   * Diode processor to log some of the action to aid in debugging
+   */
+  final class LoggingProcessor[M <: AnyRef] extends ActionProcessor[M] {
```

Comment:
I'm just curious why you do `M <: AnyRef` here. I don't think I've ever seen anyone do `M <: AnyVal` or `M <: AnyRef` before. Is it somehow important?

Reply:
That is the signature required for `ActionProcessor`. `Actions` in diode can be anything though we normally use a marker interface `Action`. So ideally this should be `LoggingProcessor[Action]` but it won't be acceptable for the call `addActionProcessor`